React client for gateway fails to compile with entities.

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

I thought this was fixed already, but it still seems to be an issue.

With an install of jhipster from master:

```
npm install -g https://github.com/jhipster/generator-jhipster.git
```
 
With the following JDL:

```
application {
  config {
    baseName uaa,
    applicationType uaa,
    packageName com.example.uaa,
    authenticationType uaa,
    uaaBaseName "uaa",
    prodDatabaseType postgresql,
    serviceDiscoveryType eureka,
    buildTool gradle
  }
}

application {
  config {
    baseName gateway,
    applicationType gateway,
    authenticationType uaa,
    packageName com.example.gateway,
    uaaBaseName "uaa",
    databaseType no,
    serviceDiscoveryType eureka,
    clientFramework react,
    buildTool gradle
  }
  entities A, B, C, D
}

application {
  config {
    baseName microservicea,
    applicationType microservice,
    authenticationType uaa,
    packageName com.example.microservicea,
    uaaBaseName "uaa",
    prodDatabaseType postgresql,
    serviceDiscoveryType eureka,
    buildTool gradle
  }
  entities A, B
}

application {
  config {
    baseName microserviceb,
    applicationType microservice,
    packageName com.example.microserviceb,
    authenticationType uaa,
    uaaBaseName "uaa",
    prodDatabaseType postgresql,
    serviceDiscoveryType eureka,
    buildTool gradle
  }
  entities C, D
}

entity A {
    data String
}

entity B {
    data String
}

entity C {
    data String
}

entity D {
    data String
}

dto * with mapstruct
service * with serviceClass

microservice A, B with microservicea
microservice C, D with microserviceb
```

When building and compiling the gateway:

```
[jsm174@mbp gateway]$ ./gradlew bootWar -Pprod jibDockerBuild

.
.
.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/a/a-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/a/a-reducer.spec.ts(232,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'IA'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/a/a-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/a/a-reducer.spec.ts(252,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'IA'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/b/b-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/b/b-reducer.spec.ts(232,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'IB'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/b/b-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microservicea/b/b-reducer.spec.ts(252,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'IB'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/c/c-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/c/c-reducer.spec.ts(232,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'IC'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/c/c-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/c/c-reducer.spec.ts(252,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'IC'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/d/d-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/d/d-reducer.spec.ts(232,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'ID'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.

ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/d/d-reducer.spec.ts
ERROR in /Users/jsm174/react-test/gateway/src/test/javascript/spec/app/entities/microserviceb/d/d-reducer.spec.ts(252,41):
TS2345: Argument of type '{ id: string; }' is not assignable to parameter of type 'ID'.
  Types of property 'id' are incompatible.
    Type 'string' is not assignable to type 'number'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! gateway@0.0.0 webpack: `node --max_old_space_size=4096 node_modules/webpack/bin/webpack.js "--config" "webpack/webpack.prod.js" "--profile"`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the gateway@0.0.0 webpack script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jsm174/.npm/_logs/2018-10-02T19_09_03_563Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! gateway@0.0.0 webpack:prod:main: `npm run webpack -- --config webpack/webpack.prod.js --profile`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the gateway@0.0.0 webpack:prod:main script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jsm174/.npm/_logs/2018-10-02T19_09_03_610Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! gateway@0.0.0 webpack:prod: `npm run cleanup && npm run webpack:prod:main`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the gateway@0.0.0 webpack:prod script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jsm174/.npm/_logs/2018-10-02T19_09_03_630Z-debug.log

> Task :webpack FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':webpack'.
> Process 'command 'npm'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 38s
4 actionable tasks: 3 executed, 1 up-to-date
```

<!-- Explain the bug, if an error is being thrown a stack trace helps -->

##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

```
gateway@0.0.0 /Users/jsm174/react-test/gateway
└── generator-jhipster@5.3.4

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "gateway",
    "clientPackageManager": "npm",
    "databaseType": "no",
    "devDatabaseType": "no",
    "enableHibernateCache": false,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": [
      "en",
      "fr"
    ],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.example.gateway",
    "packageFolder": "com/example/gateway",
    "prodDatabaseType": "no",
    "searchEngine": false,
    "serviceDiscoveryType": "eureka",
    "skipClient": false,
    "skipServer": false,
    "testFrameworks": [],
    "useSass": false,
    "websocket": false,
    "baseName": "gateway",
    "authenticationType": "uaa",
    "uaaBaseName": "uaa",
    "clientFramework": "react",
    "buildTool": "gradle",
    "jhipsterVersion": "5.3.4",
    "cacheProvider": "hazelcast",
    "serverPort": "8080",
    "skipUserManagement": false
  },
  "entities": [
    "A",
    "B",
    "C",
    "D"
  ]
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity A {
  data String
}
entity B {
  data String
}
entity C {
  data String
}
entity D {
  data String
}
dto A, B, C, D with mapstruct
service A, B, C, D with serviceClass
microservice A, B with microservicea
microservice C, D with microserviceb

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

git version 2.17.1 (Apple Git-112)

node: v10.11.0

npm: 6.4.1

Docker version 18.06.1-ce, build e68fc7a

docker-compose version 1.22.0, build f46880f

Congratulations, JHipster execution is complete!

-   [ ] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/entity-client/templates/react/src/test/javascript/spec/app/entities/entity-reducer.spec.ts.ejs
+++ b/generators/entity-client/templates/react/src/test/javascript/spec/app/entities/entity-reducer.spec.ts.ejs
@@ -276,7 +276,7 @@ describe('Entities reducer tests', () => {
         }
         <%_ } _%>
       ];
-      await store.dispatch(createEntity({ id: <% if (databaseType === 'sql') { %>1<%} else {%>'1'<%}%> })).then(() => expect(store.getActions()).toEqual(expectedActions));
+      await store.dispatch(createEntity({ id: <% if (pkType === 'Long') { %>1<%} else {%>'1'<%}%> })).then(() => expect(store.getActions()).toEqual(expectedActions));
     });
 
     it('dispatches ACTION_TYPES.UPDATE_<%= entityActionName %> actions', async () => {
@@ -297,7 +297,7 @@ describe('Entities reducer tests', () => {
         }
         <%_ } _%>
       ];
-      await store.dispatch(updateEntity({ id: <% if (databaseType === 'sql') { %>1<%} else {%>'1'<%}%> })).then(() => expect(store.getActions()).toEqual(expectedActions));
+      await store.dispatch(updateEntity({ id: <% if (pkType === 'Long') { %>1<%} else {%>'1'<%}%> })).then(() => expect(store.getActions()).toEqual(expectedActions));
     });
 
     it('dispatches ACTION_TYPES.DELETE_<%= entityActionName %> actions', async () => {

